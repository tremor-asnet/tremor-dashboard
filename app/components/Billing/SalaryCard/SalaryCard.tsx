//Libs
import { LuLandmark } from "react-icons/lu";
import { GrPaypal } from "react-icons/gr";
import { Card, Text, Flex } from "@tremor/react";

//Components
import { IconBox } from "@/components";

// Helpers
import { formattedNumber } from "@/helpers";

// Constants
import {
  CURRENCY,
  AGGREGATION_TYPE,
  AGGREGATION_DESCRIPTION,
} from "@/constants";

interface SalaryCardData {
  value: number;
  type: AGGREGATION_TYPE;
}

interface SalaryCardDataProps {
  aggregation: SalaryCardData;
}

const SalaryCard = ({ aggregation }: SalaryCardDataProps): JSX.Element => {
  const { type = aggregation.type, value = aggregation.value } = aggregation;
  const isSalary = type === AGGREGATION_TYPE.SALARY;
  const description = isSalary
    ? AGGREGATION_DESCRIPTION.SALARY
    : AGGREGATION_DESCRIPTION.PAYPAL;
  const icon = isSalary ? (
    <LuLandmark color="white" size="20px" />
  ) : (
    <GrPaypal color="white" size="18px" />
  );

  return (
    <div className="font-primary antialiased items-center justify-between py-1">
      <div className="flex items-center">
        <Card className="bg-tremor-primary dark:bg-dark-tremor-primary mx-auto pt-3 pb-4 px-4 ring-0 max-w-full lg:max-w-[356px] 2xl:max-w-full border-none relative mt-[40px] rounded-xl shadow-md">
          <Flex className="flex-col">
            <IconBox className="shadow-box-icon-primary" icon={icon} />
            <Flex className="flex-col mt-4 mb-1">
              <Text className="mb-1 text-primary dark:text-lighter text-tremor-title leading-[33px] tracking-[0.1764px] font-semibold">
                {type}
              </Text>
              <Text className="text-xs dark:text-dark-romance text-primary font-light">
                {description}
              </Text>
            </Flex>
          </Flex>
          <div className="h-px bg-[linear-gradient(to_right,rgba(52,71,103,0),rgba(52,71,103,0.4),rgba(52,71,103,0))] opacity-25 dark:opacity-95 my-4" />
          <Flex className="justify-center items-center mt-5 mb-1">
            {isSalary && (
              <Text className="text-primary text-xl h-[30px] pr-0.5 dark:text-lighter font-semibold">
                +
              </Text>
            )}
            <Text className="text-primary text-xl dark:text-lighter font-semibold">
              {formattedNumber({
                value: value,
                currency: CURRENCY.DOLLAR,
              })}
            </Text>
          </Flex>
        </Card>
      </div>
    </div>
  );
};

export default SalaryCard;
