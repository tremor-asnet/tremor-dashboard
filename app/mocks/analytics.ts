export const SALES_BY_COUNTRY = [
  {
    id: "us",
    flag: "/assets/images/flags/US.webp",
    country: "United State",
    sales: "2500",
    value: "$230,900",
    bounce: "29.9%",
  },
  {
    id: "ge",
    flag: "/assets/images/flags/DE.webp",
    country: "Germany",
    sales: "3.900",
    value: "$440,000",
    bounce: "40.22%",
  },
  {
    id: "gb",
    flag: "/assets/images/flags/GB.webp",
    country: "Great Britain",
    sales: "1.400",
    value: "$190,700",
    bounce: "23.44%",
  },
  {
    id: "br",
    flag: "/assets/images/flags/BR.webp",
    country: "Brasil",
    sales: "562",
    value: "$143,960",
    bounce: "32.14%",
  },
  {
    id: "au",
    flag: "/assets/images/flags/AU.webp",
    country: "Australia",
    sales: "400",
    value: "$143,960",
    bounce: "56.83%",
  },
];

export const STATISTICAL_DATA = [
  {
    id: "1",
    type: "Bookings",
    amount: 281,
    totalAmount: 55,
    duration: "than last week",
  },
  {
    id: "2",
    type: "Today's Users",
    amount: 2300,
    totalAmount: 3,
    duration: "than last month",
  },
  {
    id: "3",
    type: "Revenue",
    amount: 34,
    totalAmount: 1,
    duration: "than yesterday",
  },
  {
    id: "4",
    type: "Followers",
    amount: 91,
    totalAmount: 0,
    duration: "Just updated",
  },
];

export const ANALYTIC_INFO = [
  {
    id: "1",
    photo: "/assets/images/analytics/analytics-info-1.webp",
    name: "Cozy 5 Stars Apartment",
    description: `The place is close to Barceloneta Beach and bus stop just 2 min by walk and near to "Naviglio" where you can enjoy the main night life in Barcelona.`,
    price: 899,
    location: "Barcelona, Spain",
  },
  {
    id: "2",
    photo: "/assets/images/analytics/analytics-info-2.webp",
    name: "Office Studio",
    description: `The place is close to Metro Station and bus stop just 2 min by walk and near to "Naviglio" where you can enjoy the night life in London, UK.`,
    price: 1.119,
    location: "London, UK",
  },
  {
    id: "3",
    photo: "/assets/images/analytics/analytics-info-3.webp",
    name: "Beautiful Castle",
    description: `The place is close to Metro Station and bus stop just 2 min by walk and near to "Naviglio" where you can enjoy the main night life in Milan.`,
    price: 459,
    location: "Milan, Italy",
  },
];

export const WEBSITE_CHART = {
  id: "view-campaign",
  display: "Website Views",
  desc: "Last Campaign Performance",
  modified: "campaign sent 2 days ago",
  data: [
    {
      description: "M",
      sales: 50,
    },
    {
      description: "T",
      sales: 20,
    },
    {
      description: "W",
      sales: 10,
    },
    {
      description: "T",
      sales: 22,
    },
    {
      description: "F",
      sales: 50,
    },
    {
      description: "S",
      sales: 10,
    },
    {
      description: "S",
      sales: 40,
    },
  ],
};
