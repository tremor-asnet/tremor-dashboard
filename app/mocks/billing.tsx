// Constants
import { AGGREGATION_TYPE } from "@/constants";

export const mockSalaryData = [
  {
    type: AGGREGATION_TYPE.PAYPAL,
    value: 45500,
  },
  {
    type: AGGREGATION_TYPE.SALARY,
    value: 2000,
  },
];

export const mockBillingCard = {
  id: 1,
  type: "Mastercard",
  expire: "11/24",
  cardNumber: "4562 1122 4594 7866",
  holderFullName: "Jack Peterson",
  cardLast4Digit: "7866",
};
