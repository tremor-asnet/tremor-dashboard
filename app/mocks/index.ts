export * from "./card";
export * from "./profile";
export * from "./project";
export * from "./charts";
export * from "./analytics";
export * from "./sales";
export * from "./table-list";
export * from "./orders";
export * from "./product";
