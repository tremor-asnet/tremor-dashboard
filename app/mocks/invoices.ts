import { Invoice } from "@/types";

export const MOCK_INVOICES: Invoice[] = [
  {
    date: "March, 21, 2021",
    code: "MS-123456",
    price: 180,
  },
  {
    date: "Apr, 22, 2022",
    code: "MS-123457",
    price: 180,
  },
  {
    date: "May, 23, 2023",
    code: "MS-999999",
    price: 180,
  },
  {
    date: "Jun, 24, 2024",
    code: "MS-888888",
    price: 180,
  },
  {
    date: "Jul, 25, 2025",
    code: "MS-476543",
    price: 180,
  },
];
