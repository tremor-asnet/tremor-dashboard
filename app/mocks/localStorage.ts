// Mock localStorage
export const mockLocalStorage = (() => {
  let store: Record<string, string> = {};

  return {
    getItem: jest.fn((key: string) => store[key] || null),
    setItem: jest.fn((key: string, value: string) => {
      store[key] = value.toString();
    }),
    removeItem: jest.fn((key: string) => {
      delete store[key];
    }),
    clear: jest.fn(() => {
      store = {};
    }),
  };
})();

Object.defineProperty(window, "localStorage", { value: mockLocalStorage });
