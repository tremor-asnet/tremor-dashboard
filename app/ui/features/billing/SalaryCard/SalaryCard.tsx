"use client";

import { use, useState } from "react";

//Components
import { GrPaypal } from "react-icons/gr";
import { Card, Text, Flex, Button, Divider } from "@tremor/react";
import { IconBox } from "@/ui/components";

// Icons
import { MdAccountBalance } from "react-icons/md";
import { FaRegEye, FaRegEyeSlash } from "react-icons/fa6";

// Helpers
import { moneyFormat } from "@/helpers";

// Constants
import {
  CURRENCY,
  AGGREGATION_TYPE,
  AGGREGATION_DESCRIPTION,
} from "@/constants";

// Types
import { SalaryCardData } from "@/types";

import PinCode from "@/ui/features/pin-code";

const SalaryCard = ({
  type,
  value,
  currency = CURRENCY.DOLLAR,
}: SalaryCardData): JSX.Element => {
  const [isShowAmount, setIsShowAmount] = useState(false);
  const [isShowPinCodeModal, setIsShowPinCodeModal] = useState(false);

  const isSalary = type === AGGREGATION_TYPE.SALARY;

  const description = isSalary
    ? AGGREGATION_DESCRIPTION.SALARY
    : AGGREGATION_DESCRIPTION.PAYPAL;

  const icon = isSalary ? (
    <MdAccountBalance color="white" size="23px" />
  ) : (
    <GrPaypal color="white" size="18px" />
  );

  const handleConfirmSuccess = () => {
    setIsShowAmount(true);
    setIsShowPinCodeModal(false);
  };

  const toggleShowHide = () => {
    isShowAmount ? setIsShowAmount(false) : setIsShowPinCodeModal(true);
  };

  return (
    <>
      <div className="w-full font-primary antialiased items-center justify-between">
        <div className="flex items-center">
          <Card className="bg-tremor-primary dark:bg-dark-tremor-primary mx-auto pt-3 pb-4 px-4 ring-0 max-w-full lg:max-w-[356px] 2xl:max-w-full border-none relative rounded-xl shadow-md">
            <Flex flexDirection="col">
              <IconBox className="bg-gradient-pickled" icon={icon} />
              <Flex flexDirection="col" className="mt-4 mb-1">
                <Text className="mb-1 text-primary dark:text-lighter text-tremor-title leading-[33px] tracking-[0.1764px] font-semibold">
                  {type}
                </Text>
                <Text className="text-xs dark:text-dark-romance text-primary font-light">
                  {description}
                </Text>
              </Flex>
            </Flex>
            <Divider className="opacity-25 dark:opacity-15 my-4" />
            <Flex flexDirection="col" className="gap-4">
              <Button
                variant="light"
                icon={isShowAmount ? FaRegEyeSlash : FaRegEye}
                onClick={toggleShowHide}
                className="text-primary hover:text-primary dark:text-lighter dark:hover:text-lighter"
              />
              {isShowAmount ? (
                <Flex justifyContent="center">
                  {isSalary && (
                    <Text className="text-primary text-xl dark:text-lighter font-semibold">
                      +
                    </Text>
                  )}
                  <Text className="text-primary text-xl dark:text-lighter font-semibold">
                    {moneyFormat({
                      value: value,
                      currency: currency,
                    })}
                  </Text>
                </Flex>
              ) : (
                <Text className="text-primary text-xl dark:text-lighter font-semibold">
                  *****
                </Text>
              )}
            </Flex>
          </Card>
        </div>
      </div>
      <PinCode
        onConfirm={handleConfirmSuccess}
        isShowPinCodeModal={isShowPinCodeModal}
      />
    </>
  );
};

export default SalaryCard;
