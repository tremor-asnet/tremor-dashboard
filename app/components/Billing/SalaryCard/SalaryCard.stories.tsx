import type { Meta, StoryObj } from "@storybook/react";

// Components
import SalaryCard from "./SalaryCard";

//Mocks
import { mockSalaryData } from "@/mocks";

const meta = {
  title: "Components/SalaryCard",
  component: SalaryCard,
  tags: ["autodocs"],
} as Meta<typeof SalaryCard>;

export default meta;

type Story = StoryObj<typeof meta>;

export const SalaryCardDefault: Story = {
  render: () => <SalaryCard aggregation={mockSalaryData[0]} />,
};

export const SalaryCardPaypal: Story = {
  render: () => <SalaryCard aggregation={mockSalaryData[1]} />,
};
