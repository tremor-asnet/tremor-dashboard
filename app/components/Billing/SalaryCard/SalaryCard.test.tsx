import { render } from "@testing-library/react";

// Componenents
import SalaryCard from "./SalaryCard";

//Mocks
import { mockSalaryData } from "@/mocks";

describe("SalaryCard Testing", () => {
  it("should match snapshot", () => {
    const component = render(<SalaryCard salaryData={mockSalaryData[0]} />);
    expect(component).toMatchSnapshot();
  });
});
