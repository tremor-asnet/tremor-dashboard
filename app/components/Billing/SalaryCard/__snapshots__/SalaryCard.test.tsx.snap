// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SalaryCard Testing should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full font-primary antialiased items-center justify-between"
      >
        <div
          class="flex items-center"
        >
          <div
            class="tremor-Card-root w-full text-left rounded-tremor-default ring-tremor-ring shadow-tremor-card dark:ring-dark-tremor-ring dark:shadow-dark-tremor-card border-tremor-brand dark:border-dark-tremor-brand p-6 bg-tremor-primary dark:bg-dark-tremor-primary mx-auto pt-3 pb-4 px-4 ring-0 max-w-full lg:max-w-[356px] 2xl:max-w-full border-none relative rounded-xl shadow-md"
          >
            <div
              class="tremor-Flex-root flex w-full justify-between items-center flex-col"
            >
              <div
                class="tremor-Flex-root flex flex-row items-center p-1 justify-center rounded-xl w-16 h-16 bg-gradient-arsenic shadow-box-icon-primary"
              >
                <svg
                  color="white"
                  fill="currentColor"
                  height="18px"
                  stroke="currentColor"
                  stroke-width="0"
                  style="color: white;"
                  viewBox="0 0 24 24"
                  width="18px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M21.4945038,7.054125 C20.4250038,12.026625 17.0155038,14.658375 11.6050038,14.658375 L9.64300376,14.658375 L8.17975376,24.000375 L11.3620038,24.000375 C11.8217538,24.000375 12.2125038,23.665875 12.2845038,23.212125 L12.3220038,23.014125 L13.0540038,18.377625 L13.1012538,18.121875 C13.1725038,17.668125 13.5640038,17.333625 14.0230038,17.333625 L14.6042538,17.333625 C18.3640038,17.333625 21.3085038,15.805875 22.1687538,11.388375 C22.5137538,9.615375 22.3480038,8.128875 21.4945038,7.054125 M19.3172538,1.80975 C18.2057538,0.543 16.1972538,0 13.6270038,0 L6.16900376,0 C5.64325376,0 5.19625376,0.3825 5.11375376,0.90075 L2.00800376,20.59725 C1.94650376,20.98575 2.24725376,21.3375 2.64100376,21.3375 L7.24525376,21.3375 L8.40175376,14.0025 L8.36575376,14.232 C8.44825376,13.71375 8.89225376,13.33125 9.41725376,13.33125 L11.6050038,13.33125 C15.9040038,13.33125 19.2692538,11.58525 20.2525038,6.53475 C20.2817538,6.38475 20.3290038,6.0975 20.3290038,6.0975 C20.6080038,4.2285 20.3267538,2.96025 19.3172538,1.80975"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
              <div
                class="tremor-Flex-root flex w-full justify-between items-center flex-col mt-4 mb-1"
              >
                <p
                  class="mb-1 text-primary dark:text-lighter text-tremor-title leading-[33px] tracking-[0.1764px] font-semibold"
                >
                  Paypal
                </p>
                <p
                  class="text-tremor-default text-xs dark:text-dark-romance text-primary font-light"
                >
                  Freelance Payment
                </p>
              </div>
            </div>
            <div
              class="h-px bg-[linear-gradient(to_right,rgba(52,71,103,0),rgba(52,71,103,0.4),rgba(52,71,103,0))] opacity-25 dark:opacity-95 my-4"
            />
            <div
              class="tremor-Flex-root flex w-full flex-row justify-center items-center mt-5 mb-1"
            >
              <p
                class="text-tremor-default text-primary text-xl dark:text-lighter font-semibold"
              >
                $45,500
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full font-primary antialiased items-center justify-between"
    >
      <div
        class="flex items-center"
      >
        <div
          class="tremor-Card-root w-full text-left rounded-tremor-default ring-tremor-ring shadow-tremor-card dark:ring-dark-tremor-ring dark:shadow-dark-tremor-card border-tremor-brand dark:border-dark-tremor-brand p-6 bg-tremor-primary dark:bg-dark-tremor-primary mx-auto pt-3 pb-4 px-4 ring-0 max-w-full lg:max-w-[356px] 2xl:max-w-full border-none relative rounded-xl shadow-md"
        >
          <div
            class="tremor-Flex-root flex w-full justify-between items-center flex-col"
          >
            <div
              class="tremor-Flex-root flex flex-row items-center p-1 justify-center rounded-xl w-16 h-16 bg-gradient-arsenic shadow-box-icon-primary"
            >
              <svg
                color="white"
                fill="currentColor"
                height="18px"
                stroke="currentColor"
                stroke-width="0"
                style="color: white;"
                viewBox="0 0 24 24"
                width="18px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21.4945038,7.054125 C20.4250038,12.026625 17.0155038,14.658375 11.6050038,14.658375 L9.64300376,14.658375 L8.17975376,24.000375 L11.3620038,24.000375 C11.8217538,24.000375 12.2125038,23.665875 12.2845038,23.212125 L12.3220038,23.014125 L13.0540038,18.377625 L13.1012538,18.121875 C13.1725038,17.668125 13.5640038,17.333625 14.0230038,17.333625 L14.6042538,17.333625 C18.3640038,17.333625 21.3085038,15.805875 22.1687538,11.388375 C22.5137538,9.615375 22.3480038,8.128875 21.4945038,7.054125 M19.3172538,1.80975 C18.2057538,0.543 16.1972538,0 13.6270038,0 L6.16900376,0 C5.64325376,0 5.19625376,0.3825 5.11375376,0.90075 L2.00800376,20.59725 C1.94650376,20.98575 2.24725376,21.3375 2.64100376,21.3375 L7.24525376,21.3375 L8.40175376,14.0025 L8.36575376,14.232 C8.44825376,13.71375 8.89225376,13.33125 9.41725376,13.33125 L11.6050038,13.33125 C15.9040038,13.33125 19.2692538,11.58525 20.2525038,6.53475 C20.2817538,6.38475 20.3290038,6.0975 20.3290038,6.0975 C20.6080038,4.2285 20.3267538,2.96025 19.3172538,1.80975"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
            <div
              class="tremor-Flex-root flex w-full justify-between items-center flex-col mt-4 mb-1"
            >
              <p
                class="mb-1 text-primary dark:text-lighter text-tremor-title leading-[33px] tracking-[0.1764px] font-semibold"
              >
                Paypal
              </p>
              <p
                class="text-tremor-default text-xs dark:text-dark-romance text-primary font-light"
              >
                Freelance Payment
              </p>
            </div>
          </div>
          <div
            class="h-px bg-[linear-gradient(to_right,rgba(52,71,103,0),rgba(52,71,103,0.4),rgba(52,71,103,0))] opacity-25 dark:opacity-95 my-4"
          />
          <div
            class="tremor-Flex-root flex w-full flex-row justify-center items-center mt-5 mb-1"
          >
            <p
              class="text-tremor-default text-primary text-xl dark:text-lighter font-semibold"
            >
              $45,500
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
